osf/models/user.py

```diff
@@ -1508,6 +1508,56 @@ def get_node_comment_timestamps(self, target_id):
         default_timestamp = dt.datetime(1970, 1, 1, 12, 0, 0, tzinfo=pytz.utc)
         return self.comments_viewed_timestamp.get(target_id, default_timestamp)
 
+    def gdpr_delete(self):
+        """
+        This function does not remove the user object reference from our database, but it does disable the account and
+        remove identifying in a manner compliant with GDPR guidelines.
+        :return:
+        """
+        from osf.models import PreprintService, AbstractNode
+
+        user_nodes = self.nodes.all()
+        #  Validates the user isn't trying to delete things they deliberately made public.
+        if user_nodes.filter(type='osf.registration').exists():
+            raise ValidationError('You cannot delete this user because they have one or more registrations.')
```

Comment:
`ValidationError` is used when model field validation fails, so I don't think it's quite right here. A custom error might make sense, e.g. `UserStateError` (for consistency with `NodeStateError`).